tests/operators/test_python.py

```diff
@@ -817,6 +822,31 @@ def f():
 
         self._run_as_operator(f, requirements=['funcsigs', 'dill'], system_site_packages=False)
 
+    def test_private_repo_requirements(self):
+
+        with create_session() as session:
+            self._clean_stop_pypi_server()
+            self._initiate_pypi_server()
+            session.execute("delete from connection where conn_id = 'private_repo_test_conn';")
+            cn = Connection("private_repo_test_conn",
+                            "private_pypi",
+                            host="localhost",
+                            schema="http",
+                            port=8080)
+            session.add(cn)
+
+            def f():
+                import flask  # noqa: F401
+                import funcsigs  # noqa: F401
+            self._run_as_operator(
+                f,
+                requirements=['flask', 'funcsigs'],
+                connection_id="private_repo_test_conn",
+                system_site_packages=False,
+            )
+            session.commit()
+            self._clean_stop_pypi_server()
```

Comment:
When an exception is thrown this server is not stopped and will still be able to create unexpected side effects. Whay do you think about fixtures? 
See: [Teardown/Cleanup (AKA Fixture finalization)](https://docs.pytest.org/en/6.2.x/fixture.html#teardown-cleanup-aka-fixture-finalization)

Reply:
thanks will work on it